sources/packages/backend/apps/workers/src/controllers/application/application.controller.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  APPLICATION_NOT_FOUND,
+  APPLICATION_STATUS_NOT_UPDATED,
+} from "../error-code-constants";
+
+@Controller()
+export class ApplicationController {
+  constructor(
+    private readonly applicationService: ApplicationService,
+    private readonly applicationExceptionService: ApplicationExceptionService,
+  ) {}
+
+  /**
+   * Updates the student application status ensuring that the application
+   * was in the expected state and also allowing the method to be called
+   * multiple times without causing any harm to enure the impotency.
+   * @returns reports only that the job was completed without returning any
+   * output variables.
+   */
+  @ZeebeWorker("update-application-status", { fetchVariable: [APPLICATION_ID] })
+  async updateApplicationStatus(
+    job: Readonly<
+      ZeebeJob<
+        ApplicationUpdateStatusJobInDTO,
+        ApplicationUpdateStatusJobHeaderDTO,
+        IOutputVariables
+      >
+    >,
+  ): Promise<MustReturnJobActionAcknowledgement> {
+    const updateResult = await this.applicationService.updateStatus(
+      job.variables.applicationId,
+      job.customHeaders.fromStatus,
+      job.customHeaders.toStatus,
+    );
+    if (!updateResult.affected) {
+      return job.error(
+        APPLICATION_STATUS_NOT_UPDATED,
+        "The application status was not updated either because the application id was not found or the application is not in the expected status.",
+      );
+    }
+    return job.complete();
+  }
+
+  /**
+   * Searches the student application dynamic data recursively trying to
+   * find properties with the value defined as "studentApplicationException"
+   * which identifies an application exception to be reviewed by the Ministry.
+   * @returns application exceptions status.
+   */
+  @ZeebeWorker("verify-application-exceptions", {
+    fetchVariable: [APPLICATION_ID],
+  })
+  async verifyApplicationExceptions(
+    job: Readonly<
+      ZeebeJob<
+        ApplicationExceptionsJobInDTO,
+        ICustomHeaders,
+        ApplicationExceptionsJobOutDTO
+      >
+    >,
+  ): Promise<MustReturnJobActionAcknowledgement> {
+    const application = await this.applicationService.getApplicationById(
+      job.variables.applicationId,
+      { loadDynamicData: false },
+    );
+    if (!application) {
+      return job.error(APPLICATION_NOT_FOUND, "Application id not found.");
+    }
+    if (application.applicationException) {
+      // The exceptions were already processed for this application.
+      return job.complete({
+        applicationExceptionStatus:
+          application.applicationException.exceptionStatus,
+      });
+    }
+    // Check for application exceptions present in the application dynamic data.
+    const exceptions = this.applicationExceptionService.searchExceptions(
+      application.data,
+    );
+    if (exceptions.length) {
+      const createdException =
+        await this.applicationExceptionService.createException(
```

Comment:
While creating or modifying a data with workers, what will be the audit user used here?

Reply:
Not part of this PR but we need to create a solution to it. One possible solution would be to create a user for the worker, similar to the API but there is one major difference, anyone accessing our API will be authenticated, which is not true for the workers. Still open for discussion 😉

Reply:
Thanks for creating technical debt.